fecgov/fec-cms
# Schedule_a datatable search with criteria

## Summary

- Resolves #2811 
We're introducing a lot of new functionality for schedule_a datatable search, basically opening it up across all time periods as long as secondary criteria are defined.

- [x] Updated heading labels for fields to include minor heading styling
- [x] Uses multi-year select for report year box
- [x] Adds report year to the filter panel
- [x] Added custom error handling specifically for individual contributions and receipts datatables. 
        - [ ] Error message when no secondary criteria are defined: "You're searching a large dataset. Filter by recipient name or ID, contributor details, or image number."
- [x] Restricted fields are: Time periods, More contributor details, Recipient committee type, and Receipt details. These fields will lock or become disabled if multiple time periods are selected without any recipient name or ID, contributor details, or image number defined.
- [x] Path redirects with two_year_transaction_period=2020&min_date=01%2F01%2F2019&max_date=12%2F31%2F2020 in the path for both schedule_a datatables

## Impacted areas of the application
List general components of the application that this PR will affect:

- Receipts datatable
- Individual contributions datatable

## Screenshots

![Screen Shot 2019-06-13 at 10 14 51 AM](https://user-images.githubusercontent.com/12799132/59443370-dd2a7180-8dc9-11e9-9ae6-76d9faec6a48.png)

## How to test
- Check out the branch
- `npm run build`
- Go to receipts datatable: http://localhost:8000/data/receipts/ and Individual contributions datatable, http://localhost:8000/data/receipts/individual-contributions/
- Test items listed above in the Summary section of the PR. Also test for bugs and design issues and note in this PR.
____
